go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -64,7 +64,11 @@ func (bnc *BaseNetworkController) allocatePodIPsOnSwitch(pod *kapi.Pod,
 			// unknown condition how we are getting a non-terminating pod without a node here
 			klog.Errorf("Pod IP allocation found for a non-existent node in API with unknown "+
 				"condition. Pod: %s/%s, node: %s", pod.Namespace, pod.Name, pod.Spec.NodeName)
+		} else if node == nil {
```

Comment:
so `if bnc.doesNetworkRequireIPAM()` is true we want to continue with the allocation attempt and not simply return? 
or should the new check be added in line 63 above?